wallet/src/wallet/mod.rs

```diff
@@ -366,6 +427,41 @@ impl<B: storage::Backend> Wallet<B> {
         Ok(())
     }
 
+    /// Scan new blocks and update best block hash/height for the unsynced_account.
+    /// New block may reset the chain of previously scanned blocks.
+    ///
+    /// `common_block_height` is the height of the shared blocks that are still in sync after reorgs.
+    /// If `common_block_height` is zero, only the genesis block is considered common.
+    pub fn scan_new_blocks_for_unsynced_account(
+        &mut self,
+        common_block_height: BlockHeight,
+        blocks: Vec<Block>,
+    ) -> WalletResult<()> {
+        let mut db_tx = self.db.transaction_rw(None)?;
+
+        let (_, account) = self.unsynced_account.as_mut().ok_or(WalletError::NoUnsyncedAccount)?;
+
+        account.scan_new_blocks(&mut db_tx, common_block_height, &blocks)?;
+
+        db_tx.commit()?;
+
+        if account.best_block().1
+            == self
+                .accounts
+                .values()
+                .next()
+                .ok_or(WalletError::WalletNotInitialized)?
+                .best_block()
+                .1
+        {
```

Comment:
Not  a big fan of massive multi-line if conditions. Could the comparison RHS be taken out into a separate variable defined before the `if`?